src/sync/treiber_stack.rs

```diff
@@ -77,6 +77,12 @@ impl<T> TreiberStack<T> {
     }
 }
 
+impl<T> Default for TreiberStack<T> {
+    fn default() -> Self {
+        TreiberStack { head: Atomic::null() }
```

Comment:
It would be more idiomatic to leave `fn new` intact and implement `Default` by calling `TreiberStack::new()`.